### PR TITLE
Add description of topology changes to how-it-works document

### PR DESCRIPTION
**Pull request by bboreham**

### CODE REVIEW

**Comment by rade** [comment]
I don't like the ascii->html change. Markdown is meant to be readable as plain text.

**Comment by bboreham** [comment]
I have backed out the HTML change.

File: site/how-it-works.md [hunk 1]
```diff
@@ -95,6 +105,108 @@ the MAC. This way weave peers never need to exchange the MAC addresses
...
+### <a name="topology"></a>Topology Communication
+
+Topology is communicated over the TCP links between peers.
+There are two message types:
+
+##### FetchAll
+This carries no payload. The receiver responds with the entire
+topology model as the receiver has it. FetchAll is sent:
+
+ * when a new connection is established, by the peer that initiated the connection
```
Comment (rade) [inline site/how-it-works.md]: This is not correct post #349. *both* peers send FetchAll.

Comment (bboreham) [inline site/how-it-works.md]: This is deliberate, so we can document the change in #349 in a separate commit.

Comment (rade) [inline site/how-it-works.md]: I don't see the point. The doc is meant to document the state as of now. Which includes #349. As well as a whole bunch of changes over the history of the project.

File: site/how-it-works.md [hunk 2]
```diff
@@ -95,6 +105,108 @@ the MAC. This way weave peers never need to exchange the MAC addresses
...
+
+ * when a new connection is established, by both sides of the new connection
+ * when an update is received that references a peer that the receiver does not know
+ * periodically, on a timer, in case any updates have been missed
+
+##### Update
+This carries a topology payload. This is sent:
+
+  * upon receipt of a FetchAll message, as above,
+  * when a connection is added - the update will contain the two peers
```
Comment (rade) [inline site/how-it-works.md]: We need to be precise in terminology here. 'Update' is sent when the connection is marked as 'established', not when it is added.

File: site/how-it-works.md [hunk 3]
```diff
@@ -95,6 +105,108 @@ the MAC. This way weave peers never need to exchange the MAC addresses
...
+that have just connected
+  * when a connection is deleted - the update will contain just the peer
+that lost the connection
+
+The
+receiver merges it with its own topology model. If the payload is a
+subset of the receiver's topology, no further action is
+taken. Otherwise, the receiver sends out to all its connections an
+"improved" update:
```
Comment (rade) [inline site/how-it-works.md]: The above para needs re-filling.

File: site/how-it-works.md [hunk 4]
```diff
@@ -95,6 +105,108 @@ the MAC. This way weave peers never need to exchange the MAC addresses
...
+   receiver are included
+ - elements which the original payload updated in the
+   receiver are included
+ - elements which are equal between the receiver and
+   the payload are not included
+ - elements where the payload was older than the
+   receiver's version are updated
+
+If the update mentions a peer that the receiver does not know,
+then the entire update is rejected and the receiver will send a
```
Comment (rade) [inline site/how-it-works.md]: s/will send/sends

File: site/how-it-works.md (hunk 2 above)
Comment (rade) [inline site/how-it-works.md]: also, s/will contain/contains

File: site/how-it-works.md [hunk 5]
```diff
@@ -95,6 +105,108 @@ the MAC. This way weave peers never need to exchange the MAC addresses
...
+ * periodically, on a timer, in case any updates have been missed
+
+##### Update
+This carries a topology payload. This is sent:
+
+  * upon receipt of a FetchAll message, as above,
+  * when a connection is added - the update will contain the two peers
+that have just connected
+  * when a connection is deleted - the update will contain just the peer
+that lost the connection
```
Comment (rade) [inline site/how-it-works.md]: s/will contain/contains

File: site/how-it-works.md [hunk 6]
```diff
@@ -95,6 +105,108 @@ the MAC. This way weave peers never need to exchange the MAC addresses
...
+    |                ...                |
+    +-----------------------------------+
+    | Connection N: Remote Peer Name    |
+    +-----------------------------------+
+    | Connection N: Remote IP address   |
+    +-----------------------------------+
+
+#### Removal of peers
+If a peer, after receiving a topology update, sees that another peer
+no longer has any connections within the network, it will drop all
```
Comment (rade) [inline site/how-it-works.md]: s/will drop/drops

**Pull request merged**